common/flatpak-run.c

```diff
@@ -2961,8 +2961,40 @@ flatpak_run_app (const char     *app_ref,
 
   if (app_deploy != NULL)
     {
+      g_autofree const char **previous_ids = NULL;
+
       app_files = flatpak_deploy_get_files (app_deploy);
 
+      previous_ids = flatpak_deploy_data_get_previous_ids (app_deploy_data);
+      if (previous_ids)
+        {
+          for (i = 0; previous_ids[i] != NULL; i++)
+            {
+              g_autoptr (GFile) previous_data_dir = NULL;
+
+              previous_data_dir = flatpak_get_data_dir (previous_ids[i]);
+              if (g_file_query_exists (previous_data_dir, NULL))
+                {
+                  g_autoptr (GFile) current_data_dir = flatpak_get_data_dir (app_ref_parts[1]);
+                  g_autoptr (GError) local_error = NULL;
+
+                  if (flatpak_file_rename (previous_data_dir,
```

Comment:
you can ! this as well, so just do the error checking and return early if there is a problem